### ISSUE TITLE
Set up and Configuration Errors

### ISSUE THREAD

**Issue by coooolers** [brotzky/gost-zvuk#130]
1. app.use(serverConfig.handleNoToken) error
Where shall I define this function and return 401. can we define it errorHandler.js and if yes, what exactly will be the code like?

2. Where and how shall I define the Mongo URL?

3. How to setup nodemailer for mailgun?

4. yarn start and this error is there: JwtStrategy requires a secret or key
Where and how shall I set the process.env.JWT secret key? in web pack config.dev or somewhere else?

5. yarn start error:
/home/cool/Project/gost-zvuk/server/config/passport.js:62
    clientID: _config2.default.googleAuth.clientID,
  
TypeError: Cannot read property 'clientID' of undefined at passportInit (/home/cool/Project/gost-zvuk/server/config/passport.js:34:19)

Kindly explain in detail. I am a newbie.

Thanking you in anticipation

**Comment by brotzky** [comment]
Hi, if you want to get this setup locally I can explain share how the config files are supposed to look.
There are a few config files that are in the `.gitingore` as I never imagined anyone would want to run this locally :p 

This is up on github mostly so people can look around the folders if they're really bored and source control for myself. 

`.env`

```
NODE_ENV=development
MONGO_URL=mongodb://localhost:27017/{YOUR_DB_NAME}
PROTOCOL=http
HOST=localhost
APIHOST=localhost
PORT=8000
JWT={KEY}
MAIL_HOST=smtp.mailtrap.io
MAIL_PORT={KEY}
MAIL_USER={KEY}
MAIL_PASS={KEY}
```

---


`/server/config/config.js`

```
import dotenv from 'dotenv';

dotenv.config();

const config = {
  mongoURL: process.env.MONGO_URL || 'mongodb://localhost:27017/{YOUR_DB_NAME}',
  port: process.env.PORT || 8000,
  host: process.env.HOST || 'localhost',
  apiHost: process.env.APIHOST || 'localhost',
  apiPort: process.env.APIPORT || 8000,
  jwt:  {RANDOM_KEY},
  handleNoToken: (err, req, res, next) => {
    if (err.name === 'UnauthorizedError') {
      return res.status(401).send({
        data: [],
        errors: [
          {
            error: 'UNAUTHORIZED',
            message: 'Authentication required',
          },
        ],
      });
    }
    next();
  },
  googleAuth: {
    clientID: {KEY},
    clientSecret:  {KEY},
    callbackURL: `${process.env.PROTOCOL}://${process.env.HOST}:${process.env
      .PORT}/auth/google/callback`,
  },
  facebookAuth: {
    clientID:  {KEY},
    clientSecret:  {KEY},
    callbackURL: `${process.env.PROTOCOL}://${process.env.HOST}:${process.env
      .PORT}/auth/facebook/callback`,
  },
  githubAuth: {
    clientID:  {KEY},
    clientSecret: {KEY},
    callbackURL: `${process.env.PROTOCOL}://${process.env.HOST}:${process.env
      .PORT}/auth/github/callback`,
  },
  google: {
    MAPS_API_KEY: {KEY},
  },
};

export default config;
```

---

`server/mail/config.js`

```
const config = {
  host: 'smtp.mailtrap.io',
  port: {PORT},
  user: {KEY},
  pass: {KEY}
};

export default config;
```

---

I think those are all the configs. You'll have to register with the specified third parties (FB, Github, Google, mailtrap, etc) to get the keys required.

**Comment by brotzky** [comment]
> Where and how shall I define the Mongo URL?

You'll want to setup Mongo locally, create a database, and point to it while it's running in the background.

Watch some tutorials and read their docs.

This video is on Windows (I use a Mac) but goes through how to get Mongo running.
https://www.youtube.com/watch?v=pWbMrx5rVBE

**Comment by brotzky** [comment]
Yes, I'd like to build out all those features. Enough to get it usable by the public.

This is just a side project, so finding time to do everything on top of my full-time engineer job is not easy. Hopefully within a new months I can get it working

**Comment by coooolers** [comment]
![1](https://user-images.githubusercontent.com/20610084/30577636-35f938f8-9d2d-11e7-98e3-3e77fb0208c6.png)
Here is a screenshot.

Note: When I enter an address in this window, Next does not get enabled. Probably, Google Map key is not correct. I chose Google Map for JavaScript key option while creating a key from my google account.
